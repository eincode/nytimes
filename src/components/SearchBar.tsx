import React, { Component } from "react"
import { View, StyleSheet, TextInput, Image } from "react-native"
import metrics from "../config/metrics"

import IC_SEARCH from "../../assets/ic_search.png"

interface IProps {
  placeholder: string
}

export default (props: IProps) => (
  <View style={styles.container}>
    <Image source={IC_SEARCH} style={styles.icon} />
    <TextInput placeholder={props.placeholder} />
  </View>
)

const styles = StyleSheet.create({
  container: {
    width: metrics.DEVICE_WIDTH,
    height: 50,
    backgroundColor: "#FFF",
    padding: 10,
    alignItems: "center",
    flexDirection: "row",
    borderWidth: 1,
    borderColor: metrics.BORDER_COLOR,
    marginBottom: 20,
  },

  icon: {
    width: 30,
    height: 30,
    marginHorizontal: 5,
  },
})
