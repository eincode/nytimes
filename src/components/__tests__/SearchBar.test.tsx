import React from "react"
import renderer from "react-test-renderer"

import SearchBar from "../../components/SearchBar"

it("Search bar component renders correctly", () => {
  const button = renderer.create(<SearchBar />).toJSON()
  expect(button).toMatchSnapshot()
})
