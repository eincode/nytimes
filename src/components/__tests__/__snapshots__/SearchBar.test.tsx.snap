// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search bar component renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFF",
      "borderColor": "#EFEFEF",
      "borderWidth": 1,
      "flexDirection": "row",
      "height": 60,
      "padding": 10,
      "width": 750,
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../assets/ic_search.png",
      }
    }
    style={
      Object {
        "height": 30,
        "marginHorizontal": 5,
        "width": 30,
      }
    }
  />
  <TextInput
    allowFontScaling={true}
    underlineColorAndroid="transparent"
  />
</View>
`;
